### PR TITLE
Add iPXE known issue in 2.6

### PR DESCRIPTION
**Pull request by oafischer**
https://github.com/stolostron/backlog/issues/25157

### CODE REVIEW

**Review by ldpliu — APPROVED** [review]

**Comment by ldpliu** [comment]
@achuzhoy Please help to review the known issue if you have time. Thanks

**Comment by oafischer** [comment]
I talked with Vlad and will need to completely change this PR, once I've redone it I will ask him to review again.

**Comment by achuzhoy** [comment]
@ldpliu Yeah, Vlad is the person you need. And seems like it was taken care of. Thanks.

File: release_notes/known_issues.adoc [hunk 1]
```diff
@@ -564,6 +564,29 @@ compute:
...
+    matchLabels:
+      infraenvs.agent-install.openshift.io: qe2
+  bootArtifacts:
+    initrd: https://assisted-image-service-multicluster-engine.apps.qe1.kni.lab.eng.bos.redhat.com/images/73450317-b378-4f77-a3d0-e6fd98e5aefb/pxe-initrd?api_key=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiI3MzQ1MDMxNy1iMzc4LTRmNzctYTNkMC1lNmZkOThlNWFlZmIifQ.fLDph1ueGjXWQYyjWg9lPQ60mYvgQZN5KtFjRGUGdTCTEUbsWnu9A6FcnCMYVq3la3r7_ByZvWGyAvTK_M2g-g&arch=x86_64&version=4.11
+    ipxeScript: https://assisted-service-multicluster-engine.apps.qe1.kni.lab.eng.bos.redhat.com/api/assisted-install/v2/infra-envs/73450317-b378-4f77-a3d0-e6fd98e5aefb/downloads/files?api_key=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiI3MzQ1MDMxNy1iMzc4LTRmNzctYTNkMC1lNmZkOThlNWFlZmIifQ.O88mwk_WnztkmYSYmOIXwZ3fmNv_WeVmlWJm6_lZuoUaRymweFKv1TNdCurzG6pT5wtgBg7_UzV_-7d6LcyxuQ&file_name=ipxe-script
+    kernel: https://assisted-image-service-multicluster-engine.apps.qe1.kni.lab.eng.bos.redhat.com/boot-artifacts/kernel?arch=x86_64&version=4.11
+    rootfs: https://assisted-image-service-multicluster-engine.apps.qe1.kni.lab.eng.bos.redhat.com/boot-artifacts/rootfs?arch=x86_64&version=4.11
+----
+
```
Comment (vkolodny) [inline release_notes/known_issues.adoc:592]: Missing proxy part

1) Create/deploy (any) proxy server in order to expose the boot artifacts with short URLs
2) Download artifacts and copy them to the proxy
```suggestion
`for artifact in `oc get infraenv qe2 -ojsonpath="{.status.bootArtifacts}" | jq ". | keys[]" | sed "s/\"//g"`; do curl -k  `oc get infraenv qe2 -ojsonpath="{.status.bootArtifacts.${artifact}}"` -o $artifact ; done`
```
3) Copy downloaded artifacts to the proxy

File: release_notes/known_issues.adoc [hunk 2]
```diff
@@ -564,6 +564,29 @@ compute:
...
+Copy the `ipxeScript` URL into your `libvirt.xml` file.
```
Comment (vkolodny) [inline release_notes/known_issues.adoc]: Point bootp part in the libvirt.xml to the "ipxeScript"  proxy URL

File: release_notes/known_issues.adoc [hunk 3]
```diff
@@ -564,6 +564,29 @@ compute:
...
 ----
 
+[#client-cannot-reach-ipxe-script]
+=== Client cannot reach iPXE script
+
+When booting a node (such as a virtual machine when using libvirt), you might receive the following error message in the console:
+
+`no bootable devices`
+
+To work around the issue, apply an infrastructure environment custom resource definition when using an assistsed installation to expose the `bootArtifacts`, which might resemble the following file:
```
Comment (chadcrum) [inline release_notes/known_issues.adoc]: spelling - assistsed should be assisted

File: release_notes/known_issues.adoc [hunk 4]
```diff
@@ -564,6 +564,29 @@ compute:
       vmNetworkingType: 'Basic'
 ----
 
+[#client-cannot-reach-ipxe-script]
+=== Client cannot reach iPXE script
+
+When booting a node (such as a virtual machine when using libvirt), you might receive the following error message in the console:
```
Comment (vkolodny) [inline release_notes/known_issues.adoc]: Due to URL length limitation in some of DHCP servers the ipxeScript URL in infraenv will be cut and while  booting a node you might receive the following error message in the console:

File: release_notes/known_issues.adoc [hunk 5]
```diff
@@ -564,6 +564,40 @@ compute:
       vmNetworkingType: 'Basic'
 ----
 
+[#client-cannot-reach-ipxe-script]
+=== Client cannot reach iPXE script
```
Comment (swopebe) [inline release_notes/known_issues.adoc:568]: Looks great. The only question I have is does iPXE need a quick description in the beginning of the topic (not the header)? From https://ipxe.org/:

 open source network boot firmware

**Review by swopebe — APPROVED** [review]

File: release_notes/known_issues.adoc (hunk 5 above)
Comment (oafischer) [reply release_notes/known_issues.adoc:568]: Good point, I didn't see other references to iPXE in the doc so a quick description makes sense. I'll add it, thanks!

File: release_notes/known_issues.adoc [hunk 6]
```diff
@@ -564,6 +564,43 @@ compute:
...
+
+iPXE is an open source network boot firmware. See link:https://ipxe.org/[iPXE] for more details.
+
+When booting a node, the URL length limitation in some DHCP servers cuts off the `ipxeScript` URL in the infrastructure environment, resulting in the following error message in the console:
+
+`no bootable devices`
+
+To work around the issue, complete the following steps:
+
+. Apply an infrastructure environment custom resource definition when using an assisted installation to expose the `bootArtifacts`, which might resemble the following file:
```
Comment (vkolodny) [inline release_notes/known_issues.adoc]: Apply InfraEnv custom resource definition (The name of CRD is Infraenv)

Comment (oafischer) [reply release_notes/known_issues.adoc]: Fixed, thanks! If you can take one last look and approve I'll merge and close out the issue.

File: release_notes/known_issues.adoc [hunk 7]
```diff
@@ -564,6 +564,43 @@ compute:
       vmNetworkingType: 'Basic'
 ----
 
+[#client-cannot-reach-ipxe-script]
+=== Client cannot reach iPXE script
+//2.6:25157
+
+iPXE is an open source network boot firmware. See link:https://ipxe.org/[iPXE] for more details.
+
+When booting a node, the URL length limitation in some DHCP servers cuts off the `ipxeScript` URL in the infrastructure environment, resulting in the following error message in the console:
```
Comment (vkolodny) [inline release_notes/known_issues.adoc]: here the same - replace infrastructure environment with `InfraEnv`

**Review by vkolodny — COMMENTED** [review]
/lgtm

**Review by vkolodny — APPROVED** [review]

File: release_notes/known_issues.adoc [hunk 8]
```diff
@@ -564,6 +564,43 @@ compute:
...
+
+To work around the issue, complete the following steps:
+
+. Apply the `InfraEnv` custom resource definition when using an assisted installation to expose the `bootArtifacts`, which might resemble the following file:
++
+----
+status:
+  agentLabelSelector:
+    matchLabels:
+      infraenvs.agent-install.openshift.io: qe2
```
Comment (achuzhoy) [inline release_notes/known_issues.adoc:585]: You have string qe2 hardcoded. This refers to a specific domain.

File: release_notes/known_issues.adoc [hunk 9]
```diff
@@ -564,6 +564,43 @@ compute:
...
+    kernel: https://assisted-image-service-multicluster-engine.apps.qe1.kni.lab.eng.bos.redhat.com/boot-artifacts/kernel?arch=x86_64&version=4.11
+    rootfs: https://assisted-image-service-multicluster-engine.apps.qe1.kni.lab.eng.bos.redhat.com/boot-artifacts/rootfs?arch=x86_64&version=4.11
+----
+
+. Create a proxy server to expose the `bootArtifacts` with short URLs.
+
+. Copy the `bootArtifacts` and add them them to the proxy by running the following commands:
++
+----
+for artifact in oc get infraenv qe2 -ojsonpath="{.status.bootArtifacts}" | jq ". | keys[]" | sed "s/\"//g"
```
Comment (achuzhoy) [inline release_notes/known_issues.adoc:598]: You have string qe2 hardcoded. This refers to a specific domain.

File: release_notes/known_issues.adoc [hunk 10]
```diff
@@ -564,6 +564,43 @@ compute:
...
+do curl -k oc get infraenv qe2 -ojsonpath="{.status.bootArtifacts.${artifact}}"` -o $artifact
```
Comment (achuzhoy) [inline release_notes/known_issues.adoc:599]: You have string qe2 hardcoded. This refers to a specific domain.

**Pull request merged**